### PR TITLE
Feature/shopping cart

### PR DESCRIPTION
**Pull request by ArnelaKapllani**
Pls review :)
According to US7 and US8:

- Added the function to add a book in the shopping cart.
- Added the function to delete a book from the shopping cart.
- Also added the condition to check the length of the items in the shopping cart, which is not mentioned in the user story.

### CODE REVIEW

File: components/ShoppingCart/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          <div key={book.id}>
+            <BookImage book={book} />
+            <DropdownButton />
+            <DeleteButton book={book} onDelete={handleDelete} />
+            <BookmarkButton />
+          </div>
+        ))
+      ) : (
+        <p>Your cart is empty.</p>
+      )}
```
Comment (djfarly) [inline components/ShoppingCart/index.js:21]: here you'd need to look up the book from the cartItem via the complete book list

```suggestion
      {cartItems && cartItems.length > 0 ? (
        cartItems.map((cartItem) => {
          const book = books.find((book) => book.id === cartItem.bookId);
          if (!book) {
            return null;
          }
          return (
            <div key={book.id}>
              <BookImage book={book} />
              <DropdownButton />
              <DeleteButton book={book} onDelete={handleDelete} />
              <BookmarkButton />
            </div>
          );
        })
      ) : (
        <p>Your cart is empty.</p>
      )}
```

**Review by djfarly — CHANGES_REQUESTED** [review]
Hej Arnela,

great work so far! ✨ 

Even though this is not QA, I was kinda confused why I could not add a book to the cart from the overview page or the details page.

I've added some notes about the state structure for your cart. They certainly are not complete but maybe they paint a picture how you could build it.

Keep up the good work! 👍

**Comment by ArnelaKapllani** [comment]
> Hej Arnela,
> 
> great work so far! ✨
> 
> Even though this is not QA, I was kinda confused why I could not add a book to the cart from the overview page or the details page.
> 
> I've added some notes about the state structure for your cart. They certainly are not complete but maybe they paint a picture how you could build it.
> 
> Keep up the good work! 👍

Thanks for your Review @djfarly 🙏 In fact, I thought about this, but I didn't want to make the buying option from everywhere, this also to shorten the work for me 🙈 If you suggest that it will be better to do it, then I will change it.

File: components/ShoppingCart/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+export default function ShoppingCart({ cartItems, handleDelete }) {
+  return (
+    <>
+      <h1>My cart</h1>
+      {cartItems && cartItems.length > 0 ? (
+        cartItems.map((book) => (
+          <div key={book.id}>
+            <BookImage book={book} />
+            <DropdownButton />
```
[the remark this answers is not in the document]
Comment (ArnelaKapllani) [reply components/ShoppingCart/index.js:14]: You mean the Dropdown component? Yes, i will modify it :)

File: pages/_app.js [hunk 3]
```diff
@@ -1,12 +1,47 @@
...
 
 export default function App({ Component, pageProps }) {
+  const [cartItems, setCartItems] = useState([]);
+
+  function addToCart(bookId) {
+    const existigCartItem = cartItems.find(
+      (cartItem) => cartItem.bookId === bookId
+    );
+    if (existigCartItem) {
+      if (existigCartItem.quantity < 5) {
```
Comment (ekat-br) [inline pages/_app.js:13]: Just to understand it better: Why exactly is this condition necessary? (current quantity is less than 5..)

**Review by ekat-br — APPROVED** [review]

**Comment by ekat-br** [comment]
Hi @ArnelaKapllani,  the desired functions work and the corresponding code looks good to me. Well done! I have left you with a small question of understanding.

File: pages/_app.js (hunk 3 above)
Comment (ArnelaKapllani) [reply pages/_app.js:13]: Because i added the condition of not buying more than 5 books. It could be < 100 e.g. 🙂

Comment (ekat-br) [reply pages/_app.js:13]: Oh, that means I can add more than 5 books to the shopping cart, but I can't pay for more than 5. Have I understood that correctly?

Comment (ArnelaKapllani) [reply pages/_app.js:13]: No, it means you cannot add more than 5 copies of the book to your shopping cart.

Comment (ekat-br) [reply pages/_app.js:13]: Ah ok it's about the copies of a single book. Now it all makes sense, thank you!

Comment (ArnelaKapllani) [reply pages/_app.js:13]: 😊

**Pull request merged**